### PR TITLE
Use upstream RPC-MaaS (liberty)

### PR DESCRIPTION
**Pull request by cloudnull**
This change converts our deployment of RPC MaaS from the old role method
to the new playbook method. This change will provide us access to a
faster and more flexible monitoring solution and allow all customers to
benifit from Monitoring as a Service using a unified code base.

The maas group_vars have been removed and the values of the old file
have been placed into the `user_osa_variables_defaults.yml` file. This
has been done to allow the rpc-maas playbooks to inherit from the user
variables in cases where the group vars not not available.

Signed-off-by: Kevin Carter <kevin.carter@rackspace.com>
(cherry picked from commit ca573c173a0d62cef3ee0a2e463e02b8d0160c9f)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
Hmm, the artifact jobs are not supposed to be running against this branch. Let me look into that. Note that they can safely be ignored.

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
Given that the jobs did not re-run, it seems that these somehow got kicked off in some sort of interim state. The following job results can be ignored completely:

#. CIT/artifact-xenial-Apt
#. CIT/artifact-xenial-Container
#. CIT/artifact-xenial-Git
#. CIT/artifact-xenial-Python
#. CIT/issue-link
#. CIT/xenial-deploy-swift

**Comment by mattt416** [comment]
@odyssey4me I think the problem lies here:

```
cloudnull changed the base branch to rcbops:liberty-12.2 from rcbops:master 14 days ago
```

@cloudnull, you may want to just close this PR and re-open it against liberty-12.2 so that all the correct jobs are triggered.

**Comment by odyssey4me** [comment]
The user_*_defaults.yml file was introduced in mitaka, so it does not apply to kilo/liberty.

File: scripts/deploy.sh [hunk 1]
```diff
@@ -158,6 +158,16 @@ fi
 cd ${RPCD_DIR}/playbooks
 openstack-ansible -i "localhost," patcher.yml
 
+# set permissions and lay down overrides files
+chmod 0440 /etc/openstack_deploy/user_*_defaults.yml
+if [[ ! -f "${RPCD_OVERRIDES}" ]]; then
+  cp "${RPCD_DIR}/etc/openstack_deploy/user_rpco_variables_overrides.yml" "${RPCD_OVERRIDES}"
```
Comment (evrardjp) [inline scripts/deploy.sh]: This probably needs to go

File: scripts/bootstrap-ansible.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# All updates (security and otherwise) must come from the RPC-O apt artifacting.
+#
+# This is being done via bash because Ansible is not bootstrapped yet, and the
+# apt artifacts used for bootstrapping Ansible must also come from the RPC-O
+# artifact repo.
+#
+# This has the ability to be disabled for the purpose of reusing the
+# bootstrap-ansible script for putting together the apt artifacts.
+if [[ "${HOST_SOURCES_REWRITE}" == 'yes' ]] && apt_artifacts_available; then
+  configure_apt_sources
```
Comment (evrardjp) [inline scripts/bootstrap-ansible.sh]: I don't think we should do that.

File: scripts/deploy.sh (hunk 1 above)
Comment (cloudnull) [inline scripts/deploy.sh]: done.

**Comment by cloudnull** [comment]
@evrardjp which interface do you want to port over?

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_trusty_deploy_swift

File: rpcd/etc/openstack_deploy/user_variables.yml [hunk 3]
```diff
@@ -40,6 +40,7 @@ nova_db_max_pool_size: 120
 nova_db_pool_timeout: 60
 nova_cpu_allocation_ratio: 2.0
 nova_ram_allocation_ratio: 1.0
+nova_console_type: "spice"
```
Comment (evrardjp) [inline rpcd/etc/openstack_deploy/user_variables.yml]: that should not be in user_variables.
If you need it for rpc maas, it should be in rpcm variables. But I strongly doubt we need this.

Comment (cloudnull) [inline rpcd/etc/openstack_deploy/user_variables.yml]: This is required in the earlier releases as maas can not interpret the role defaults. This is following the already merged pattern in kilo (https://github.com/rcbops/rpc-openstack/blob/kilo/rpcd/etc/openstack_deploy/user_variables.yml#L41)

**Pull request merged**